examples/datagen_with_al/filter_gen_data.py

```diff
@@ -0,0 +1,141 @@
+import logging
+from enum import Enum
+from operator import attrgetter, itemgetter
+from typing import List
+
+from examples.datagen_with_al.utils.data import LMFilter, RTFilter, unpack_samples
+from primeqa.mrc.models.heads.extractive import EXTRACTIVE_HEAD
+from primeqa.mrc.models.task_model import ModelForDownstreamTasks
+from primeqa.mrc.processors.postprocessors.scorers import SupportedSpanScorers
+from primeqa.mrc.processors.postprocessors.squad import SQUADPostProcessor
+from transformers import AutoConfig, AutoTokenizer, DataCollatorWithPadding, TrainingArguments
+
+from datasets import concatenate_datasets, load_from_disk
+
+format_str = "[%(levelname)s - %(name)s - %(asctime)s] %(message)s"
+datefmt_str = "%d-%m-%Y %H:%M:%S"
+logging.basicConfig(format=format_str, datefmt=datefmt_str, level=logging.INFO)
+logger = logging.getLogger()
+
+
+class Strategy(Enum):
+    LM = "lm"
+    RT = "rt"
+
+    def __str__(self) -> str:
+        return self.value
+
+
+def filter_samples(
```

Comment:
Please add doc string and inline comments. Thanks.

Reply:
Sorry forgot it for this script, next push will include them!